src/arc.ts

```diff
@@ -586,6 +599,90 @@ export function pollForAccountChanges(currentAccountAddress: Address | null, int
   });
 }
 
+async function monitorGraphNodeSubgraph() {
+  if (targetedNetwork() === "ganache") {
+    const query = `{
+      subgraphs {
+        name
+        currentVersion {
+          deployment {
+            id
+            latestEthereumBlockNumber
+            totalEthereumBlocksCount
+            synced
+            failed
+          }
+        }
+      }
+    }`;
+
+    try {
+      const subgraphURL = settings.ganache.graphqlHttpProvider.split("/name/daostack")[0];
+      const { data } = (await axios.post(subgraphURL, { query })).data;
+      for (const i in data.subgraphs) {
+        if (data.subgraphs[i].name === "daostack") {
+          const { id, failed, synced, latestEthereumBlockNumber } = data.subgraphs[i].currentVersion.deployment;
+          return { id, failed, synced, latestEthereumBlockNumber };
+        }
+      }
+    } catch (e) {
+      console.error("Failed in monitoring graph node", e);
+      return { id: null, failed: true, synced: false, latestEthereumBlockNumber: 0 };
+    }
+  }
+
+  const subgraphName = (settings as any)[process.env.NETWORK]["graphqlHttpProvider"].split("https://api.thegraph.com/subgraphs-daostack/name/daostack/")[1];
+  const query = `{
+    indexingStatusesForSubgraphName(subgraphName: "daostack/` + subgraphName + `") { subgraph synced failed chains { network ... on EthereumIndexingStatus { latestBlock { number hash } chainHeadBlock { number hash } } } }
+  }`;
+
+  try {
+    const { data } = (await axios.post("https://api.thegraph.com/index-node/graphql", { query })).data;
```

Comment:
index-node ?

Reply:
hey i took this code from the alchemybots repo, and it works, so if there is a different way to monitor thegraph then let me know